### PR TITLE
feat(Metrics Router): add new Metrics Router service to project

### PR DESCRIPTION
**Pull request by adya1303**
## PR summary
This PR adds the Metrics Router service to the project, including integration tests and request examples.

## PR Checklist
Please make sure that your PR fulfills the following requirements:  
- [ ] The commit message follows the [Angular Commit Message Guidelines](https://github.com/angular/angular/blob/master/CONTRIBUTING.md#-commit-message-guidelines).
- [x] Tests for the changes have been added (for bug fixes / features)
- [ ] Docs have been added / updated (for bug fixes / features)

## Does this PR introduce a breaking change?    
- [ ] Yes
- [x] No

### CODE REVIEW

**Comment by adya1303** [comment]
**unit tests**
```
~/Documents/GO/src/github.com/IBM/platform-services-go-sdk/metricsrouterv3 (regen-mr-061222) $ go test
Running Suite: MetricsRouterV3 Suite
====================================
Random Seed: 1670477696
Will run 91 of 91 specs

•••••••••••••••••••••••••••••••••••••••••••••••••••••••••••••••••••••••••••••••••••••••••••
Ran 91 of 91 Specs in 6.766 seconds
SUCCESS! -- 91 Passed | 0 Failed | 0 Pending | 0 Skipped
```

**integration tests**
```
~/Documents/GO/src/github.com/IBM/platform-services-go-sdk/metricsrouterv3 (regen-mr-061222) $ go test -tags="integration" -v
=== RUN   TestMetricsRouterV3
Running Suite: MetricsRouterV3 Suite
====================================
Random Seed: 1670478288
Will run 135 of 135 specs

•••••••••••••
------------------------------
• [SLOW TEST:5.289 seconds]
MetricsRouterV3 Integration Tests
/Users/adyanandjha/Documents/GO/src/github.com/IBM/platform-services-go-sdk/metricsrouterv3/metrics_router_v3_integration_test.go:42
  ListTargets - List targets
  /Users/adyanandjha/Documents/GO/src/github.com/IBM/platform-services-go-sdk/metricsrouterv3/metrics_router_v3_integration_test.go:392
    ListTargets(listTargetsOptions *ListTargetsOptions)
    /Users/adyanandjha/Documents/GO/src/github.com/IBM/platform-services-go-sdk/metricsrouterv3/metrics_router_v3_integration_test.go:397
------------------------------
•••••••••••••••••••••••••••••••••••••••••••••••••••••••••••••••••••••••••••••••••••••••••••••••••••••••••••••••••••••••••
Ran 135 of 135 Specs in 28.251 seconds
SUCCESS! -- 135 Passed | 0 Failed | 0 Pending | 0 Skipped
```

**example tests**
```
~/Documents/GO/src/github.com/IBM/platform-services-go-sdk/metricsrouterv3 (regen-mr-061222) $ go test -tags="examples" -v
=== RUN   TestMetricsRouterV3
Running Suite: MetricsRouterV3 Suite
====================================
Random Seed: 1670478421
Will run 106 of 106 specs

•••••••••••••••••••••••••••••••••••••••••••••••••••••••••••••••••••••••••••••••••••••••••••••
Ran 106 of 106 Specs in 11.220 seconds
SUCCESS! -- 106 Passed | 0 Failed | 0 Pending | 0 Skipped
```

**Comment by padamstx** [comment]
@adya1303 In preparation for a formal SF review of the SDK work in early January, could you please update your branch with the latest changes from "main", and when you are already for a review please click the "Ready for review" button on your draft PR.  Thanks!

**Comment by adya1303** [comment]
> @adya1303 In preparation for a formal SF review of the SDK work in early January, could you please update your branch with the latest changes from "main", and when you are already for a review please click the "Ready for review" button on your draft PR. Thanks!



> @adya1303 In preparation for a formal SF review of the SDK work in early January, could you please update your branch with the latest changes from "main", and when you are already for a review please click the "Ready for review" button on your draft PR. Thanks!

sure @padamstx

**Comment by adya1303** [comment]
@padamstx Can you please review the PR. Thanks

File: metricsrouterv3/metrics_router_v3.go
Comment (padamstx) [inline metricsrouterv3/metrics_router_v3.go:45]: This was generated as the default service URL because it was the first entry in the "servers" field within the API definition.  Is this the desired default endpoint for the service?  If not, then please adjust the order of the entries within the "servers" field so that the entry containing the default endpoint URL is the first entry in the list.

**Review by padamstx — APPROVED** [review]
LGTM, just need to verify that the sydney region is the correct default endpoint URL (see my separate comment on that).

**Comment by padamstx** [comment]
I've approved the PR and I will merge when it is the appropriate time to do so. This would be at about the same time as the service implementation is made available in the production environment.  Just let me know when and I will merge.

**Comment by padamstx** [comment]
@adya1303 Could you please update your branch with the latest changes from "main"?
Also, please address my question about the default endpoint URL above.  Thanks.

File: metricsrouterv3/metrics_router_v3.go
Comment (adya1303) [reply metricsrouterv3/metrics_router_v3.go:45]: For this GA we are supporting only au-syd, so the DefaultServiceURL is good for now.

**Comment by adya1303** [comment]
> @adya1303 Could you please update your branch with the latest changes from "main"? Also, please address my question about the default endpoint URL above. Thanks.

Updated the branch with master and replied to the comment. For this GA, au-syd is the only supported region, so DefaultServiceURL is ok.

**Comment by padamstx** [comment]
@adya1303 I just merged latest changes in main again.  Is it time to merge the PR if the build is clean?

**Comment by padamstx** [comment]
We'll hold this PR out until @adya1303 gives the go-ahead, later this quarter.

**Comment by padamstx** [comment]
I did a re-review and the changes in the PR look good.
Before we merge this PR (IIRC the ETA is May/June when the service is GA), we'll need to make sure the PR is updated using the current "main" branch at the time, and we'll need screenshots showing clean results from running integration tests and examples.  We can defer this until shortly before we plan to merge.

**Comment by adya1303** [comment]
Tested with the latest code changes - 
**unit tests**
```
~/Documents/GO/src/github.com/IBM/platform-services-go-sdk/metricsrouterv3 (add-mr-12132022) $ go test                       
Running Suite: MetricsRouterV3 Suite
====================================
Random Seed: 1685023090
Will run 83 of 83 specs

•••••••••••••••••••••••••••••••••••••••••••••••••••••••••••••••••••••••••••••••••••
Ran 83 of 83 Specs in 2.945 seconds
SUCCESS! -- 83 Passed | 0 Failed | 0 Pending | 0 Skipped
PASS
ok  	github.com/IBM/platform-services-go-sdk/metricsrouterv3	4.087s
```
**integration tests**
```
~/Documents/GO/src/github.com/IBM/platform-services-go-sdk/metricsrouterv3 (add-mr-12132022) $ go test -tags="integration" -v
=== RUN   TestMetricsRouterV3
Running Suite: MetricsRouterV3 Suite
====================================
Random Seed: 1685022846
Will run 123 of 123 specs
<some tests results output>
•••••••••••••••••••••••••••••••••••••••••••••••••••••••••••••••••••••••••••••••••••••••••••••••••••••••••••••••••••••••••••
Ran 123 of 123 Specs in 15.119 seconds
SUCCESS! -- 123 Passed | 0 Failed | 0 Pending | 0 Skipped
```
**example tests**
```
~/Documents/GO/src/github.com/IBM/platform-services-go-sdk/metricsrouterv3 (add-mr-12132022) $ go test -tags="examples" -v
=== RUN   TestMetricsRouterV3
Running Suite: MetricsRouterV3 Suite
====================================
Random Seed: 1685022970
Will run 97 of 97 specs

••••••••••••••••••••••••••••••••••••••••••••••••••••••••••••••••••••••••••••••••••••••
Ran 97 of 97 Specs in 5.395 seconds
SUCCESS! -- 97 Passed | 0 Failed | 0 Pending | 0 Skipped

**Comment by adya1303** [comment]
Hi, @padamstx, we are ready to merge the changes. Tested with the current code and have added the screenshots.
Could you merge the changes. Thanks

**Pull request merged**
